src/private.h

```diff
@@ -113,6 +113,19 @@ struct _gpuarray_blas_ops {
   int (*setup)(gpucontext *ctx);
   void (*teardown)(gpucontext *ctx);
   const char *(*error)(gpucontext *ctx);
+
+  int (*hdot)( size_t N,
+    gpudata *X, size_t offX, int incX,
+    gpudata *Y, size_t offY, int incY,
```

Comment:
It seems some BLAS functions are using `int` as stride (such as `*gemv *ger`) while others are using `size_t`. For now I just made a revert to `size_t`. I'm experimenting with `ssize_t` however getting test errors. Still need some work to finish.